es74x-cloud/src/main/scala/tech/beshu/ror/proxy/es/clients/HighLevelClientBasedClusterAdminClient.scala

```diff
@@ -0,0 +1,279 @@
+/*
+ *     Beshu Limited all rights reserved
+ */
+package tech.beshu.ror.proxy.es.clients
+
+import monix.execution.Scheduler
+import org.elasticsearch.action._
+import org.elasticsearch.action.admin.cluster.allocation._
+import org.elasticsearch.action.admin.cluster.health._
+import org.elasticsearch.action.admin.cluster.node.hotthreads._
+import org.elasticsearch.action.admin.cluster.node.info._
+import org.elasticsearch.action.admin.cluster.node.reload.NodesReloadSecureSettingsRequestBuilder
+import org.elasticsearch.action.admin.cluster.node.stats._
+import org.elasticsearch.action.admin.cluster.node.tasks.cancel._
+import org.elasticsearch.action.admin.cluster.node.tasks.get._
+import org.elasticsearch.action.admin.cluster.node.tasks.list._
+import org.elasticsearch.action.admin.cluster.node.usage._
+import org.elasticsearch.action.admin.cluster.repositories.cleanup._
+import org.elasticsearch.action.admin.cluster.repositories.delete._
+import org.elasticsearch.action.admin.cluster.repositories.get._
+import org.elasticsearch.action.admin.cluster.repositories.put._
+import org.elasticsearch.action.admin.cluster.repositories.verify._
+import org.elasticsearch.action.admin.cluster.reroute._
+import org.elasticsearch.action.admin.cluster.settings._
+import org.elasticsearch.action.admin.cluster.shards._
+import org.elasticsearch.action.admin.cluster.snapshots.create._
+import org.elasticsearch.action.admin.cluster.snapshots.delete._
+import org.elasticsearch.action.admin.cluster.snapshots.get._
+import org.elasticsearch.action.admin.cluster.snapshots.restore._
+import org.elasticsearch.action.admin.cluster.snapshots.status._
+import org.elasticsearch.action.admin.cluster.state._
+import org.elasticsearch.action.admin.cluster.stats._
+import org.elasticsearch.action.admin.cluster.storedscripts._
+import org.elasticsearch.action.admin.cluster.tasks._
+import org.elasticsearch.action.ingest._
+import org.elasticsearch.action.support.master.AcknowledgedResponse
+import org.elasticsearch.client.ClusterAdminClient
+import org.elasticsearch.cluster.{ClusterName, ClusterState}
+import org.elasticsearch.common.bytes.BytesReference
+import org.elasticsearch.common.xcontent.XContentType
+import org.elasticsearch.tasks.TaskId
+import org.elasticsearch.threadpool.ThreadPool
+import tech.beshu.ror.proxy.es.exceptions.NotDefinedForRorProxy
+
+class HighLevelClientBasedClusterAdminClient(esClient: RestHighLevelClientAdapter)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Number of methods in class exceeds 30](https://app.codacy.com/app/scarduzio/elasticsearch-readonlyrest-plugin/pullRequest?prid=4591272)